src/psyclone/psyGen.py

```diff
@@ -5815,3 +5910,47 @@ def gen_c_code(self, indent=0):
         :rtype: str
         '''
         return self._value
+
+
+class Return(Node):
+    '''
+    Node representing a Return statement (subroutine break without return
```

Comment:
Fortran is unusual in that it does not permit a return value to be supplied to the return statement. Since other languages do and this is an IR, should we future-proof ourselves here and allow for an optional return value or did you have some other plan for that functionality?

Reply:
As you say, Fortran does not have return with value and neither OpenCL (__kernels are always void type), so we don't need it for now. My feeling is that if we parse a language that has return value syntax we could specify this in the symbol table and then use an assignment to this return value + the routine exit return.